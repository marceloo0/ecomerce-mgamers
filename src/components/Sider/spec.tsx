import { screen } from '@testing-library/react'
import { renderWithTheme } from '../../utils/tests/helpers'

import { Sider } from '.'

const handleNextSlide = jest.fn()
const handlePrevSlide = jest.fn()

describe('<Sider />', () => {
  it('should render default', () => {
    const { container } = renderWithTheme(
      <Sider
        handleNextSlide={handleNextSlide}
        handlePrevSlide={handlePrevSlide}
      />
    )

    expect(screen.getByLabelText(/back/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
