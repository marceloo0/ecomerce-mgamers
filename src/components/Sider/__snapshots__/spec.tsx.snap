// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sider /> should render default 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 1px solid transparent;
  border-radius: 0.4rem;
  cursor: pointer;
  color: #C7ECEC;
  background-color: #201C2B;
  padding: 0.5rem;
  background-color: transparent;
}

.c1:hover {
  background-color: #332756;
}

.c1 > svg {
  color: #7FFA45;
}

.c1:hover {
  border: 1px solid #7FFA45;
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 12px;
  width: 100%;
  height: 180px;
}

<div
  class="c0"
>
  <button
    aria-label="back"
    class="c1"
  >
    <svg
      fill="currentColor"
      height="36"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
      />
    </svg>
  </button>
  <button
    class="c1"
  >
    <svg
      fill="currentColor"
      height="36"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="36"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
      />
    </svg>
  </button>
</div>
`;
